contribs/desync-resources.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+# Copyright 2021 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import argparse
+import csv
+import io
+import shutil
+import sys
+
+from nestbox_confd_client import Client as ConfdClient
+from wazo_auth_client import Client as AuthClient
+
+
+def _extract_verify_certificate(value):
+    if value.lower() == 'true':
+        return True
+    if value.lower() == 'false':
+        return False
+    return value
+
+
+def main():
+    args = parse_args()
+    verify_certificate = _extract_verify_certificate(args.verify_certificate)
+
+    auth_client = AuthClient(
+        args.host,
+        port=443,
+        prefix='/api/auth',
+        verify_certificate=verify_certificate,
+        username=args.username,
+        password=args.password,
+    )
+    token = auth_client.token.new()['token']
+    auth_client.set_token(token)
+
+    confd_client = ConfdClient(
+        args.host,
+        port=443,
+        prefix='/api/confd',
+        verify_certificate=verify_certificate,
+        token=token,
+    )
+    rcl = []
+    if args.rcl:
+        rcl = detect_desync_rcl(auth_client, confd_client)
+
+    users = []
+    if args.users:
+        force_synchro = args.delete_orphan_users
+        users = detect_desync_users(auth_client, confd_client, force_synchro)
+
+    if not rcl and not users:
+        sys.exit(0)
+
+    result = io.StringIO()
+    fieldnames = ['uuid', 'type', 'name']
+    writer = csv.DictWriter(result, fieldnames=fieldnames)
+    writer.writeheader()
+    for row in rcl + users:
+        writer.writerow(row)
+
+    if args.output:
+        with open(args.output, 'w') as fobj:
+            result.seek(0)
+            shutil.copyfileobj(result, fobj)
+    else:
+        print(result.getvalue())
+
+    result.close()
+    sys.exit(1)
+
+
+def detect_desync_users(auth_client, confd_client, force_synchro):
+    result = []
+    response = auth_client.users.list(recurse=True)
+    users = {user['uuid']: user for user in response['items']}
+
+    confd_users = confd_client.users.list()['items']
+    for confd_user in confd_users:
+        user = users.get(confd_user['uuid'])
+        if not user:
+            result.append({
+                'uuid': confd_user['uuid'],
+                'type': 'user',
+                'name': '',
+            })
+            if force_synchro:
```

Comment:
The deletion step should not be mixed with the detection step, and the boolean argument `force_synchro` is bleh :)